src/lang/std/sketch.ts

```diff
@@ -1895,13 +1896,131 @@ export function replaceSketchLine({
+  edgeCutMeta: EdgeCutInfo | null
+):
+  | {
+      modifiedAst: Program
+      tag: string
+    }
+  | Error {
+  const _node = structuredClone(tagInfo.node)
+  let pipeIndex = 0
+  for (let i = 0; i < tagInfo.pathToNode.length; i++) {
+    if (tagInfo.pathToNode[i][1] === 'PipeExpression') {
+      pipeIndex = Number(tagInfo.pathToNode[i + 1][0])
+      break
+    }
+  }
+  const pipeExpr = getNodeFromPath<PipeExpression>(
+    _node,
+    tagInfo.pathToNode,
+    'PipeExpression'
+  )
+  if (err(pipeExpr)) return pipeExpr
+  const callExpr = pipeExpr.node.body[pipeIndex]
+  if (callExpr.type !== 'CallExpression')
+    return new Error('no chamfer call Expr')
+  const obj = callExpr.arguments[0]
+  if (obj.type !== 'ObjectExpression')
+    return new Error('first argument should be an object expression')
+  const tags = obj.properties.find((a) => {
+    return a.key.name === 'tags'
+  })
+  if (!tags) return new Error('no tags property')
+  if (tags.value.type !== 'ArrayExpression')
+    return new Error('tags should be an array expression')
+  if (tags.value.elements.length < 2) {
+    return addTag(2)(tagInfo)
+  }
+  const tagIndexToPullOut = tags.value.elements.findIndex((element) => {
+    if (
+      edgeCutMeta?.subType === 'base' &&
+      element.type === 'Identifier' &&
+      element.name === edgeCutMeta.tagName
+    )
+      return true
+    if (
+      edgeCutMeta?.subType === 'opposite' &&
+      element.type === 'CallExpression' &&
+      element.callee.name === 'getOppositeEdge' &&
+      element.arguments[0].type === 'Identifier' &&
+      element.arguments[0].name === edgeCutMeta.tagName
+    )
+      return true
+    if (
+      edgeCutMeta?.subType === 'adjacent' &&
+      element.type === 'CallExpression' &&
+      (element.callee.name === 'getNextAdjacentEdge' ||
+        element.callee.name === 'getPrevAdjacentEdge') &&
+      element.arguments[0].type === 'Identifier' &&
+      element.arguments[0].name === edgeCutMeta.tagName
+    )
+      return true
+    return false
+  })
+  if (tagIndexToPullOut === -1) return new Error('tag not found')
+  const tagToPullOut = tags.value.elements[tagIndexToPullOut]
+  tags.value.elements.splice(tagIndexToPullOut, 1)
+  const chamferLength = obj.properties.find(
+    (a) => a.key.name === 'length'
+  )?.value
+  if (!chamferLength) return new Error('no chamfer length')
+  const tagDec = createTagDeclarator(findUniqueName(_node, 'seg', 2))
+  const newExpressionToInsert = createCallExpression('chamfer', [
+    createObjectExpression({
+      length: chamferLength,
+      tags: createArrayExpression([tagToPullOut]),
+    }),
+    createPipeSubstitution(),
+    tagDec,
+  ])
+  pipeExpr.node.body.splice(pipeIndex, 0, newExpressionToInsert)
+  return {
+    modifiedAst: _node,
+    tag: tagDec.value,
+  }
+}
+
 export function addTagForSketchOnFace(
   tagInfo: AddTagInfo,
-  expressionName: string
-) {
+  expressionName: string,
+  edgeCutMeta: EdgeCutInfo | null
```

Comment:
you've probably explained it already, sorry if asking again, but what is edge cut meta ?

Reply:
![image](https://github.com/user-attachments/assets/9b6b9c99-478b-469d-be07-04fdac04d09e)

It's some metadata that needed to know how to break up the chamfer call expressions

i.e. 
```ts
chamfer({
       length: 18,
       tags: [getNextAdjacentEdge(seg01), seg02]
     }, %)
```
needs to be broken into two chamfer calls for each of the tags in the array, so that a tag declorator can be added to one of them, but some extra information is needed to know which one needs the tag declorator.